### PR TITLE
Refactored Lease and tenants to use shared API model | psp 2977

### PR DESCRIPTION
**Pull request by FuriousLlama**

### CODE REVIEW

File: frontend/src/features/leases/detail/LeasePages/tenant/Tenant.tsx [hunk 1]
```diff
@@ -1,17 +1,103 @@
...
 import styled from 'styled-components';
+import { getContactMethodValue } from 'utils/contactMethodUtil';
 import { withNameSpace } from 'utils/formUtils';
+import { formatApiPersonNames } from 'utils/personUtils';
 
 import TenantNotes from './TenantNotes';
 import TenantOrganizationContactInfo from './TenantOrganizationContactInfo';
 import TenantPersonContactInfo from './TenantPersonContactInfo';
 
+class FormAddress {
```
Comment (FuriousLlama) [inline frontend/src/features/leases/detail/LeasePages/tenant/Tenant.tsx:20]: Necessary classes to match the underlying form requirements.

File: backend/api/Startup.cs [hunk 2]
```diff
@@ -106,6 +106,7 @@ public void ConfigureServices(IServiceCollection services)
             });
             services.Configure<JsonSerializerOptions>(options =>
             {
+                options.ReferenceHandler = ReferenceHandler.IgnoreCycles;
```
Comment (FuriousLlama) [inline backend/api/Startup.cs:109]: This is new with dotnet6. It allows the json to work even with recurcive structures by ignoring them.

File: backend/api/Models/Concepts/Organization/OrganizationPersonMap.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+using Entity = Pims.Dal.Entities;
+
+namespace Pims.Api.Models.Concepts
+{
+    public class OrganizationPersonMap : IRegister
+    {
+        public void Register(TypeAdapterConfig config)
+        {
+            config.NewConfig<Entity.PimsPersonOrganization, OrganizationPersonModel>()
+                .PreserveReference(true)
```
Comment (FuriousLlama) [inline backend/api/Models/Concepts/Organization/OrganizationPersonMap.cs:11]: Keeps the references when mapping

File: backend/api/Controllers/ModelController.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        /// <returns></returns>
+        [HttpGet("{id:long}")]
+        [HasPermission(Permissions.LeaseView)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(Pims.Api.Models.Concepts.PersonModel), 200)]
+        [SwaggerOperation(Tags = new[] { "person" })]
+        public IActionResult GetPerson(int id)
+        {
+            return new JsonResult(id);
+        }
```
Comment (asanchezr) [inline backend/api/Controllers/ModelController.cs:42]: Is this code meant to be part of the PR? Seems to be a placeholder result...

Comment (FuriousLlama) [reply backend/api/Controllers/ModelController.cs:42]: partly, this provides a dummy route that generates a separate page for swagger and the new models. I used it to test the generation of the api models though swagger

**Review by asanchezr — APPROVED** [review]

**Pull request merged**